presto-native-execution/presto_cpp/main/SessionProperties.h

```diff
@@ -0,0 +1,192 @@
+  /// writing to disk for IO efficiency.
+  static constexpr const char* kSpillWriteBufferSize =
+      "native_spill_write_buffer_size";
+
+  /// Config used to create spill files. This config is provided to underlying
+  /// file system and the config is free form.
+  static constexpr const char* kSpillFileCreateConfig =
+      "native_spill_file_create_config";
+
+  /// Enable window spilling on native engine.
+  static constexpr const char* kWindowSpillEnabled =
+      "native_window_spill_enabled";
+
+  /// Enable writer spilling on native engine.
+  static constexpr const char* kWriterSpillEnabled =
+      "native_writer_spill_enabled";
+
+  /// The number of bits (N) used to calculate the spilling
+  /// partition number for hash join and RowNumber: 2 ^ N
+  static constexpr const char* kJoinSpillPartitionBits =
+      "native_join_spiller_partition_bits";
+
+  static constexpr const char* kNativeSpillerNumPartitionBits =
+      "native_spiller_num_partition_bits";
+
+  /// Enable topN row number spilling on native engine.
+  static constexpr const char* kTopNRowNumberSpillEnabled =
+      "native_topn_row_number_spill_enabled";
+
+  /// If set to true, then during execution of tasks, the output vectors of
+  /// every operator are validated for consistency. This is an expensive check
+  /// so should only be used for debugging.
+  static constexpr const char* kValidateOutputFromOperators =
+      "native_debug_validate_output_from_operators";
+
+  /// Enable timezone-less timestamp conversions.
+  static constexpr const char* kLegacyTimestamp = "legacy_timestamp";
+
+  /// Specifies the cpu time slice limit in ms that a driver thread
+  /// can continuously run without yielding.
+  static constexpr const char* kDriverCpuTimeSliceLimitMs =
+      "driver_cpu_time_slice_limit_ms";
+
+  SessionProperties();
+
+  std::shared_ptr<SessionProperty> getSessionProperty(
+      const std::string& name) const;
+
+  const std::unordered_map<std::string, std::shared_ptr<SessionProperty>>&
+  getSessionProperties() const;
+
+  /// Utility function to translate a config name in Presto to its equivalent in
+  /// Velox. Returns 'name' as is if there is no mapping.
+  std::string toVeloxConfig(const std::string& name) const;
+
+ protected:
+  void addSessionProperty(
+      const std::string& name,
+      const std::string& description,
+      const velox::TypePtr& type,
+      bool isHidden,
+      const std::string& veloxConfigName,
+      const std::optional<std::string> veloxDefault);
+
+  std::unordered_map<std::string, std::shared_ptr<SessionProperty>>
+      sessionProperties_;
+};
+
+class SessionPropertyReporter {
+ public:
+  SessionPropertyReporter() {
+    sessionProperties_ = std::make_shared<SessionProperties>();
+  }
+
+  json getSessionPropertyMetadata(const std::string& name);
+
+  json getSessionPropertiesMetadata();
+
+ protected:
+  std::shared_ptr<SessionProperties> sessionProperties_;
```

Comment:
Does this need to be a shared_ptr ? If this object is only used within the class, then it can be just an object, it doesn't need a shared_ptr.